apps/src/fish/FishView.jsx

```diff
@@ -53,23 +47,98 @@ class FishView extends React.Component {
     mobilePortraitWidth: PropTypes.number.isRequired
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height(),
+      appWidth: $('#codeApp').width(),
+      appHeight: $('#codeApp').height()
+    };
+  }
+
   componentDidMount() {
     this.props.onMount();
+
+    this.setState({
```

Comment:
do we need to set this same state in both the constructor and componentDidMount?

Reply:
Just `componentDidMount()` should be fine, though `render()` will be called once before that happens (which is probably fine, you just can't count on those state values being present)

Reply:
If the window gets resized, we'd run into problems, right?

Reply:
Oh, sorry - missed the event listener below

Reply:
Yes, I set them here so that the first render would succeed.

Reply:
Now it looks like you are setting them in the constructor as well. just curious, why the change?

Reply:
Oh, my comment was misplaced.  I set them in the constructor so that they work before the first render.  

Of all places, setting them here after the mount might be overkill, though given how much testing we just gave this change, I'm nervous to remove it without a fresh round of testing.